src/server/pfs/drive/driver.go

```diff
@@ -1071,6 +1072,11 @@ func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint6
 	if fileInfo.FileType == pfs.FileType_FILE_TYPE_NONE {
 		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
 	}
+	// We return NotFound if it's a regular file but its blocks do not match the
+	// given filter shard
+	if fileInfo.FileType == pfs.FileType_FILE_TYPE_REGULAR && len(blockRefs) == 0 && (filterShard != nil && filterShard.BlockNumber > 1) {
+		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
+	}
```

Comment:
I think the check that the `filterShard` here is none `nil` and that `BlockNumber > 1` is redundant. It doesn't seem it should be possible to have `FILE_TYPE_REGULAR` and no `blockRefs` unless we eliminated some of them due to a `filterShard`. Is that true or is there some way to hit that code?